lib/RGBot.js

```diff
@@ -932,101 +1654,167 @@ const RGBot = class {
                             amountToWithdraw = slot.count > amountToWithdraw ? amountToWithdraw : slot.count;
-                            await containerWindow.withdraw(slot.type, null, amountToWithdraw);
+                            await openContainer.withdraw(slot.type, null, amountToWithdraw);
                             quantityCollected += amountToWithdraw;
                         }
                     }
+                    quantityCollectedMap[slot.name] = quantityCollected
                 }
             }
+            return Object.entries(quantityCollectedMap).find(q => q[1] > 0)?.value || false;
         }
+        return false
     }
 
     /**
      * Deposits one or more items into a container.
      * @param {Window} containerWindow The open container Window to deposit items into
-     * @param {object} options={} Optional parameters
+     * @param {object} [options={}] Optional parameters
      * @param {string} [options.itemName=undefined] An Item to deposit into the container. If not specified, will deposit all Items.
      * @param {boolean} [options.partialMatch=false] Allow partial matches to itemName. For example, 'planks' will match any Item containing 'planks' in its name ('spruce_planks', 'oak_planks', etc.).
      * @param {number} [options.quantity=undefined] If itemName is specified, deposit up to this quantity.
      * @returns {Promise<void>}
+     *
+     * @deprecated Since 1.2.0, RGBot now manages state while using a container/chest as documented on field isLookingInContainer, please use openAndUseContainer(options, depositItemsToContainer)
      */
     async depositItems(containerWindow, options = {}) {
-        const itemName = options.itemName || undefined;
+        this.#logWarn(`depositItems(containerWindow, options): deprecated Since 1.2.0, RGBot now manages state while using a container/chest as documented on field isLookingInContainer, please use openAndUseContainer(options, depositItemsToContainer)`)
+        options['itemNames'] = options.itemName && [options.itemName] || []
+        await this.depositItemsToContainer(containerWindow, options)
+    }
+
+    /**
+     * Should be passed as the `useContainerFunction` to openAndUseContainer.   Deposits one or more items into the container.
+     * @param {Window} openContainer The open container Window
+     * @param {object} [options={}] Optional parameters
+     * @param {string} [options.itemNames=[]] The items to deposit into the container. If not specified, will deposit all Items.
+     * @param {boolean} [options.partialMatch=false] Allow partial matches to itemNames. For example, '_planks' will match any Item containing '_planks' in its name ('spruce_planks', 'oak_planks', etc.).
+     * @param {number} [options.quantity=undefined] If itemNames is specified, deposit up to this quantity of each itemName.
+     * @returns {Promise<boolean>}
+     */
+    async depositItemsToContainer(openContainer, options = {}) {
```

Comment:
Same here. General comment on these functions - in my strategies, I only ever want to put items into an operation container... so why should I have to open the container myself? I think a single function that handles both opening and placing items or withdrawing items would be great to have.

Reply:
After totally re-writing this, I went back to design almost like what Abby started with, except it supports list arguments.

The reason is that there is a different 'open' method for each container type in mineflayer.. yes.. really